replay/src/execution.rs

```diff
@@ -0,0 +1,284 @@
+
+    let tx = get_blockifier_transaction(
+        reader,
+        block_context.block_info().block_number,
+        tx_hash,
+        execution_flags,
+    )
+    .inspect_err(|err| error!("failed to fetch transaction: {}", err))?;
+
+    info!("starting transaction execution");
+
+    let pre_execute_instant = Instant::now();
+    let execution_result = tx.execute(state, block_context);
+    let execution_time = pre_execute_instant.elapsed();
+
+    // TODO: Move this to the caller.
+    // This function should only execute the transaction and return relevant information.
+    #[cfg(feature = "state_dump")]
+    crate::state_dump::create_state_dump(
+        state,
+        block_context.block_info().block_number.0,
+        &tx_hash.to_hex_string(),
+        &execution_result,
+    );
+
+    // TODO: Move this to the caller.
+    // This function should only execute the transaction and return relevant information.
+    #[cfg(feature = "with-libfunc-profiling")]
+    crate::libfunc_profile::create_libfunc_profile(tx_hash.to_hex_string());
+
+    match &execution_result {
+        Ok(execution_info) => {
+            let result_string = execution_info
+                .revert_error
+                .as_ref()
+                .map(|err| err.to_string())
+                .unwrap_or("ok".to_string());
+
+            info!("transaction execution finished: {}", result_string);
+
+            let receipt = reader.get_tx_receipt(tx_hash)?;
+            validate_tx_with_receipt(execution_info, receipt);
+        }
+        Err(err) => error!("transaction execution failed: {err}"),
+    }
+
+    Ok(TransactionExecution {
+        result: execution_result,
+        time: execution_time,
+        hash: tx_hash,
+    })
+}
+
+/// Validates the transaction execution with the network receipt,
+/// and logs the result
+fn validate_tx_with_receipt(
+    execution_info: &TransactionExecutionInfo,
+    receipt: RpcTransactionReceipt,
+) {
+    #[derive(PartialEq, Debug, Serialize)]
+    struct TransactionSummary {
+        reverted: bool,
+        events: usize,
+        messages: usize,
+    }
+
+    let actual_execution_summary = {
+        TransactionSummary {
+            reverted: execution_info.is_reverted(),
+            events: execution_info
+                .receipt
+                .resources
+                .starknet_resources
+                .archival_data
+                .event_summary
+                .n_events
+                + 1,
+            messages: execution_info
+                .receipt
+                .resources
+                .starknet_resources
+                .messages
+                .l2_to_l1_payload_lengths
+                .len(),
+        }
+    };
+    let expected_execution_summary = {
+        TransactionSummary {
+            reverted: receipt.execution_result.revert_reason().is_some(),
+            events: receipt.events.len(),
+            messages: receipt.messages_sent.len(),
+        }
+    };
```

Comment:
Is it necessary to have the creation of the struct inside `{ }`? Couldn´t it be like this?:

```suggestion
    let expected_execution_summary = TransactionSummary {
            reverted: receipt.execution_result.revert_reason().is_some(),
            events: receipt.events.len(),
            messages: receipt.messages_sent.len(),
   };
```
Not really a problem, just made me curious